go/libraries/doltcore/schema/typeinfo/null.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package typeinfo
+
+import (
+	"github.com/src-d/go-mysql-server/sql"
+
+	"github.com/liquidata-inc/dolt/go/store/types"
+)
+
+type nullImpl struct{}
```

Comment:
So it's possible to set a column type to `types.NullKind`, and with this being an eventual complete replacement for `NomsKind`, we have to be able to set a column to a `Null` type. The only way out of this would be to internally remove all instances where we are setting a column to `types.NullKind`, and I'm not sure how many places that is in.